engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/ModuleScope.java

```diff
@@ -196,10 +198,24 @@ public Function lookupMethodDefinition(AtomConstructor atom, String name) {
     if (definedWithAtom != null) {
       return definedWithAtom;
     }
+
     Function definedHere = getMethodMapFor(atom).get(lowerName);
     if (definedHere != null) {
       return definedHere;
     }
+
+    if (atom.isBuiltin()) {
+      // Unfortunately locally defined extensions get associated with the module rather than a type
```

Comment:
I don't understand this comment or its relevance to your work – why are you looking for methods on the defined module?

Reply:
To explain my confusion:
```
type File
    type File
    
    foo = 123
```
works just fine inside `File.enso` – the method gets defined on the atom, not the module. So what's the difference with `@Builtin_Method`?

Reply:
I believe this was the problem with the case like
```
File.foo = 123
```
but I will double check

Reply:
Yeap. So let me elaborate because this is probably a known feature which just happens to blow in my face now because I moved types likes Nothing and Any to stdlib.

Consider a module Foo defined `src/Data/Foo.enso`:
```
type Foo

    type Foo a

    plus_one  = 1 + a

new x = Foo x
```

And now you define extension methods on Foo
```
from Standard.Base.Data.Foo import all

Foo.two = 2

Foo.Foo.three = 3

main =
    x = Foo.new 0
    x.two // not OK
    x.there // fine
```

This is how it works even in the current master.
What I'm fixing with that condition is a typical scenario encountered in our tests and stdlib:
```
from Standard.Base.Data.Any import all

Any.method = 1
main = 3.method // will fail with method not found
```
you can fix it with
```
Any.Any.method = ...
```
but I assumed that this is not desireable. But writing about it now... maybe it is.
Maybe we can take it offline.

Reply:
So after my changes it looks like the full qualifier is only present in tests, which I would say is acceptable. And as we mentioned, this will go away with statics.